src/daemon/control.rs

```diff
@@ -456,6 +543,80 @@ fn share_unvault_signatures(
     send_sig_msg(&mut transport, txid, sigs.clone())
 }
 
+// Will panic if not called by a manager
+fn fetch_cosigner_signatures(
+    revaultd: &RevaultD,
+    spend_tx: &mut SpendTransaction,
+) -> Result<(), Box<dyn std::error::Error>> {
+    for (host, noise_key) in revaultd.cosigs.as_ref().expect("We are manager").iter() {
+        // FIXME: connect should take a reference... This copy is useless
+        let mut transport = KKTransport::connect(*host, &revaultd.noise_secret, &noise_key)?;
+        let msg = SignRequest {
+            tx: spend_tx.clone(),
+        };
+        log::debug!(
+            "Sending '{}' to cosigning server at '{}' (key: '{:?}')",
+            &serde_json::to_string(&msg)?,
+            host,
+            noise_key
+        );
+        transport.write(&serde_json::to_vec(&msg)?)?;
+
+        let res_msg: SignResponse = serde_json::from_slice(&transport.read()?)?;
+        log::debug!(
+            "Receiving '{}' from cosigning server",
+            &serde_json::to_string(&res_msg)?,
+        );
+
+        // FIXME: i abuse jsonrpc_core::Error here to avoid creating YA Error struct when we are
+        // going to actually start throwing JSONRPC errors in this thread soon!
+        let res_tx = res_msg.tx.ok_or(jsonrpc_core::Error::invalid_params(
+            "One of the Cosigning Server already signed a Spend transaction spending \
+                one of these vaults!"
+                .to_string(),
+        ))?;
+        if res_tx.inner_tx().inputs.len() != spend_tx.inner_tx().inputs.len() {
+            // They lied to us, no need to go further.
+            log::error!(
+                "Cosigning server sent us a bad Spend transaction: '{}', our is '{}'",
+                res_tx.as_psbt_string(),
+                spend_tx.as_psbt_string()
+            );
+            return Ok(());
```

Comment:
Can you please explain why you're returning Ok(()) even if the cosigner server lied?

Reply:
To be honest i don't think we can have decent detailed errors until we get rid of the JSONRPC thread and stuff calls in jsonrpc/api.rs

This is just going to fail the finalization later, i know it's horrible..

Reply:
i mean i can just abuse jsonrpc::Error again..

Reply:
Oh, actually it can not happen https://github.com/revault/revault_tx/blob/b0502d50ac20d65fbc5f4960d3aae7f7e736da69/src/transactions.rs#L480-L487

Reply:
```diff
diff --git a/src/daemon/control.rs b/src/daemon/control.rs
index b377788..84603ba 100644
--- a/src/daemon/control.rs
+++ b/src/daemon/control.rs
@@ -579,22 +579,15 @@ fn fetch_cosigner_signatures(
                 one of these vaults!"
                 .to_string(),
         ))?;
-        if res_tx.inner_tx().inputs.len() != spend_tx.inner_tx().inputs.len() {
-            // They lied to us, no need to go further.
-            log::error!(
-                "Cosigning server sent us a bad Spend transaction: '{}', our is '{}'",
-                res_tx.as_psbt_string(),
-                spend_tx.as_psbt_string()
-            );
-            return Ok(());
-        }
 
         for (i, psbtin) in res_tx.into_psbt().inputs.into_iter().enumerate() {
             spend_tx
                 .inner_tx_mut()
                 .inputs
                 .get_mut(i)
-                .expect("Checked above")
+                .expect(
+                    "A SpendTransaction cannot have a different number of txins and PSBT inputs",
+                )
                 .partial_sigs
                 .extend(psbtin.partial_sigs);
         }
```